java-squid/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -672,7 +680,19 @@ boolean isInheritedIn(JavaSymbol symbol, JavaSymbol.TypeJavaSymbol clazz) {
       case Flags.PUBLIC:
         return true;
       case Flags.PRIVATE:
-        return symbol.owner() == clazz;
+        // FIXME take only constructors into account when all the members should be JLS-6.6.1
+        if ("<init>".equals(symbol.name)) {
+          JavaSymbol topLevel1 = symbol;
+          JavaSymbol topLevel2 = clazz;
+          while (!topLevel1.owner.isPackageSymbol()) {
+            topLevel1 = topLevel1.owner;
+          }
+          while (!topLevel2.owner.isPackageSymbol()) {
+            topLevel2 = topLevel2.owner;
+          }
+          return topLevel1 == topLevel2;
+        }
+        return symbol.owner == clazz;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1698)